parser/SqlParser.ypp

```diff
@@ -543,6 +555,15 @@ create_table_statement:
     $$ = new quickstep::ParseStatementCreateTable(@1.first_line, @1.first_column, $3, $5);
   };
 
+create_index_statement:
```

Comment:
Having `CREATE INDEX` as a separate statement might be a bit tricky because of the way that index sub-blocks are handled in Quickstep. Since indices are internal to blocks and not external and global, it is difficult to add indices in an ad-hoc fashion if a relation already has blocks in it. We could modify the default StorageBlockLayout so that newly-created blocks have the appropriate indices, but dealing with pre-existing blocks is a trickier issue. We could either:

1. Leave existing blocks as-is without the new indices. This is lightweight and easy to implement but may confound users' expectations.
2. Reorganize the entirety of the existing relation with the new layout (essentially bulk-copying blocks into blocks of the new layout with a trivial SELECT operation). This is potentially quite heavyweight but ensures that all existing data is indexed. Another wrinkle is how we should deal with blocks that may not have the default layout for the relation (it's always been a goal for Quickstep, and something that is supported in the storage engine, to have different blocks in a relation have different layouts, although we don't have good support for that in the default SQL query path yet).

One option that sidesteps this issue entirely is to handle creating indices using a sub-clause of the `CREATE TABLE` statement. That way, we're always establishing indices as part of the default layout for a table up-front.